src/queries/flatRateFulfillmentMethods.js

```diff
@@ -15,7 +15,24 @@ export default async function flatRateFulfillmentMethods(context, input) {
 
   await context.validatePermissions("reaction:legacy:shippingMethods", "read", { shopId });
 
-  return Shipping.find({
-    shopId
-  });
+  // aggregate pipeline to extract fulfillment methods inside shippment
+  return {
+    collection: Shipping,
+    pipeline: [
+      {
+        $match: {
+          "provider.name": "flatRates",
+          shopId
+        }
+      },
+      {
+        $unwind: "$methods"
+      },
+      {
+        $replaceRoot: {
+          newRoot: "$methods"
+        }
+      }
```

Comment:
I discovered that there's an error when you request `shop` field because `shopId` is on the original root document. To fix it, please change this last `replaceRoot` stage to:

```js
{
        $replaceRoot: {
          newRoot: {
            $mergeObjects: [
              "$methods",
              {
                shopId: "$$ROOT.shopId"
              }
            ]
          }
        }
      }
```